src/json/mapping.cr

```diff
@@ -60,12 +60,12 @@ module JSON
   # If *strict* is `true`, unknown properties in the JSON
   # document will raise a parse exception. The default is `false`, so unknown properties
   # are silently ignored.
-  macro mapping(properties, strict = false)
-    {% for key, value in properties %}
-      {% properties[key] = {type: value} unless value.is_a?(HashLiteral) || value.is_a?(NamedTupleLiteral) %}
+  macro mapping(_properties_, strict = false)
```

Comment:
Does %properties work for macro arguments?

Reply:
Nope, syntax error.

Reply:
That wouldn't make sense, because the macro argument `properties` is a macro-level variable, whereas `%`-prefixed variables are local variables in the generated code. This are two different levels.

Reply:
Well, it obviously doesn't work, but it is unfortunate that it is impossible to avoid collisions.